src/commands/transaction/cosign.ts

```diff
@@ -58,33 +59,33 @@ export default class extends ProfileCommand {
         const account = profile.decrypt(options);
         const accountHttp = new AccountHttp(profile.url);
         const transactionHttp = new TransactionHttp(profile.url);
+        let transactionWasFound = false;
+
         const hash = new HashResolver()
             .resolve(options, undefined, '\'Enter the aggregate bonded transaction hash to cosign: ');
 
-        this.getGraphAccounts(profile)
-            .pipe(
-                mergeMap((_) => _),
-                mergeMap((address) => accountHttp.getAccountPartialTransactions(address, new QueryParams(100))),
-                mergeMap((_) => _),
-                filter((_) => _.transactionInfo !== undefined && _.transactionInfo.hash !== undefined &&
-                    _.transactionInfo.hash === hash), // Filter transaction
-                toArray(),
-            )
-            .subscribe((transactions: AggregateTransaction[]) => {
-                if (!transactions.length) {
-                    this.spinner.stop(true);
-
-                    let text = '';
-                    text += chalk.red('Error');
-                    console.log(text, 'The profile', profile.name, 'cannot cosign the transaction with hash', hash, '.');
-                } else {
-
-                    const transaction = transactions[0];
+        const networkCall = (address: Address) => accountHttp
+            .getAccountPartialTransactions(address, new QueryParams(100))
+            .toPromise();
 
-                    const cosignatureTransaction = CosignatureTransaction.create(transaction);
-                    const signedCosignature = account.signCosignatureTransaction(cosignatureTransaction);
+        const sequentialFetcher = SequentialFetcher.create(networkCall);
 
-                    transactionHttp.announceAggregateBondedCosignature(signedCosignature).subscribe(
+        this.getSelfAndChildrenAddresses(profile)
+            .pipe(
+                switchMap((addresses) => sequentialFetcher.getTransactionsToCosign(addresses)),
+                flatMap((transaction) => transaction),
+                filter((_) => _.transactionInfo !== undefined
+                    && _.transactionInfo.hash !== undefined
+                    && _.transactionInfo.hash === hash),
+            )
+            .subscribe((transaction: AggregateTransaction) => {
+                sequentialFetcher.kill();
+                const cosignatureTransaction = CosignatureTransaction.create(transaction);
+                const signedCosignature = account.signCosignatureTransaction(cosignatureTransaction);
+                transactionWasFound = true;
+                transactionHttp
+                    .announceAggregateBondedCosignature(signedCosignature)
+                    .subscribe(
```

Comment:
I don't like the idea of nesting two subscribes. We could use mergeMap/flatMap instead.

Reply:
I don't recommend to do that since it would make error management tricky, as there are 5 possible outcomes there (see below)

I split the functions to make it more readable, please tell me if you want me to modify it

Things that can go wrong
1. getSelfAndChildrenAddresses
2. getTransactionsToCosign
3. signCosignatureTransaction
4. announceAggregateBondedCosignature
... 5. OK outcome